src/components/data-grid/export/docDefinition.ts

```diff
@@ -0,0 +1,250 @@
+import { ColumnApi, GridApi, ValueFormatterParams } from '@ag-grid-community/core';
+import { Content, Margins, TDocumentDefinitions } from 'pdfmake/interfaces';
+import { PdfHeaderCell, PdfTableCell, PrintParams, Translations } from '../types';
+
+export const getDocDefinition = (
+  gridApi: GridApi,
+  columnApi: ColumnApi,
+  printParams: PrintParams,
+  translations: Translations,
+) => {
+  const {
+    PDF_HEADER_COLOR,
+    PDF_INNER_BORDER_COLOR,
+    PDF_OUTER_BORDER_COLOR,
+    PDF_ODD_BKG_COLOR,
+    PDF_EVEN_BKG_COLOR,
+    PDF_HEADER_HEIGHT = 25,
+    PDF_ROW_HEIGHT = 15,
+    PDF_PAGE_ORIENTATION,
+    PDF_WITH_HEADER_IMAGE,
+    PDF_WITH_FOOTER_PAGE_COUNT,
+    PDF_LOGO = '',
+  } = printParams;
+
+  return (function () {
+    const columnGroupsToExport = getColumnGroupsToExport();
+
+    const columnsToExport = getColumnsToExport();
+
+    const widths = getExportedColumnsWidths(columnsToExport);
+
+    const rowsToExport = getRowsToExport(columnsToExport);
+
+    const body = columnGroupsToExport
+      ? [columnGroupsToExport, columnsToExport, ...rowsToExport]
+      : [columnsToExport, ...rowsToExport];
+
+    const headerRows = columnGroupsToExport ? 2 : 1;
+
+    const header = PDF_WITH_HEADER_IMAGE
+      ? ({
+          image: 'ag-grid-logo',
+          width: 150,
+          alignment: 'center',
+          margin: [0, 10, 0, 10],
+        } as Content)
+      : undefined;
+
+    const footer = PDF_WITH_FOOTER_PAGE_COUNT
```

Comment:
suggesting name `getFooter` due to shadow-scoping the var name `footer` (if I used the term name correctly lol)